sys/newlib/syscalls.c

```diff
@@ -186,7 +186,7 @@ int _open_r(struct _reent *r, const char *name, int flags, int mode)
  *
  * @return      TODO
  */
-int _read_r(struct _reent *r, int fd, void *buffer, unsigned int count)
+_ssize_t _read_r(struct _reent *r, int fd, void *buffer, size_t count)
```

Comment:
I think it's just because newlib doesn't want to pollute the namespace and only defines names starting with underscore. There's a typedef in the cross toolchain headers `typedef ssize_t _ssize_t`.